src/ui/map.js

```diff
@@ -1043,20 +1047,43 @@ class Map extends Camera {
      * @see [Example: Display a web map using an alternate projection](https://docs.mapbox.com/mapbox-gl-js/example/map-projection/)
      * @see [Example: Use different map projections for web maps](https://docs.mapbox.com/mapbox-gl-js/example/projections/)
      */
-    setProjection(projection?: ?ProjectionSpecification | string) {
+    setProjection(projection: ProjectionSpecification | string = {name: "mercator"}) {
         this._lazyInitEmptyStyle();
         if (typeof projection === 'string') {
             projection = (({name: projection}: any): ProjectionSpecification);
         }
-        this._runtimeProjection = projection;
-        this._transitionFromGlobe = false;
-        this.style.updateProjection();
+        this.updateProjection(projection);
         return this;
     }
 
-    _setImplicitProjection(projection: ProjectionSpecification) {
-        this._runtimeProjection = projection;
-        this.style.updateProjection();
+    updateProjection(projection?: ProjectionSpecification) {
+        if (projection) { this._runtimeProjection = projection; }
+        const prevProjectionState = this.transform.projection;
+        // newProjection is a user-friendly projection object or null if not changed
+        const newProjection = this.transform.setProjection(this._runtimeProjection || (this.style.stylesheet ? this.style.stylesheet.projection : undefined));
+        const projectionState = this.transform.projection;
+
+        this.style.tryDraping();
```

Comment:
nit: I would rename that function since `tryDraping` suggests the action of draping, so it almost sounds like some draping rendering will happen when called. What this function does is trying to enable terrain to later _leverage_ draping.

Reply:
Good catch @karimnaaji!